docs/target-setup-es2k.md

```diff
@@ -0,0 +1,116 @@
+this custom program are pre-generated in the package provided by Intel.
+Please use these artifacts and put them under k8s_dp/es2k dir of k8s source.
+If any modifications are made, use the following instructions on
+compilation under [Compile K8s P4](#compile-k8s-p4)
+section of this guide. The default cp_init.cfg file would need changes for subfunction support
+on the host. An example file called `cp_init_use_case_cdq.cfg` has been provided.
+
+Copy fxp-net_k8s-dp.pkg to IMC '/work/scripts' dir from the link partner machine.
+
+  ```bash
+  scp fxp-net_k8s-dp.pkg 100.0.0.100:/work/scripts/.
+  ```
+
+  On IMC:
+  Modify load_custom_pkg.sh as below:
+
+  ```bash
+  cd /work/scripts
+  cp /etc/dpcp/cp_init_use_case_cdq.cfg cp_init.cfg
+  ```
+
+  ```bash
+  cat load_custom_pkg.sh
+  ```
+  Modify the script `load_custom_pkg.sh`
+  so that it looks like the below
+
+  ```bash
+  #!/bin/sh
+  CP_INIT_CFG=/etc/dpcp/cfg/cp_init.cfg
+  echo "Checking for custom package..."
+  if [ -e fxp-net_k8s-dp.pkg ]; then
+      echo "Custom package p4_custom.pkg found. Overriding default package"
+      cp  fxp-net_k8s-dp.pkg /etc/dpcp/package/
+      cp  cp_init.cfg /etc/dpcp/cfg/
+      rm -rf /etc/dpcp/package/default_pkg.pkg
+      ln -s /etc/dpcp/package/fxp-net_k8s-dp.pkg /etc/dpcp/package/default_pkg.pkg
+      sed -i 's/sem_num_pages = 1;/sem_num_pages = 25;/g' $CP_INIT_CFG
+      sed -i 's/lem_num_pages = 1;/lem_num_pages = 25;/g' $CP_INIT_CFG
+  else
+      echo "No custom package found. Continuing with default package"
+  fi
+  ```
+
+The work directory on IMC is persistent so any config and package files
+copied here, will continue to exist over subsequent reboots.
+
+## Set up Host
+
+Follow the steps listed in the FXP P4 SDE User Guide document for host setup.
+The steps include configuring Linux kernel boot parameters to
+enable IOMMU, ATE, etc.; installing ATE kernel and other RPMs; installing
+required third-party software; installing p4sde and p4-cp-nws.
+
+## Install out-of-tree IDPF Driver
+
+IDPF driver creates subfunction interfaces which are allocated to Pods.
+Please refer to the IDPF documentation on how to install (or build if required for
+the right version). Documentation under features/networking/IDPF_Readme.rst in the 
+documentation tarball
+
+## Install IPDK SDE and IPDK Networking Recipe
+
+For K8s recipe on host, IPDK p4-cp-nws (p4-control) and p4-sde components need to be installed
+and run on the host natively.
+To install p4sde, follow the instructions in FXP P4SDE User Guide.
+
+To install infrap4d which is the networking recipe, follow instructions at
+https://github.com/ipdk-io/networking-recipe/blob/main/docs/guides/setup/es2k-setup-guide.md
+Make sure to checkout the appropriate
+
+## Install P4C
+
+P4 PNA compiler is used to build P4 compiled artifacts. The source distribution
+for CPT and P4C is under in the p4-programs tarball hw-p4-programs.xxxx.tgz.
+
+## Compile K8s P4
```

Comment:
Also add  "The P4 Programs User Guide describes how to build packages and
artifacts for a sample P4 program. See the **Build custom package**
chapter in the Intel P4 Programs Readme for detailed instructions." on top